subsys/bluetooth/services/ots/ots_dir_list.c

```diff
@@ -208,7 +208,18 @@ void bt_ots_dir_list_init(struct bt_ots_dir_list **dir_list, void *obj_manager)
 {
 	struct bt_gatt_ots_object *dir_list_obj;
 	int err;
+
+	__ASSERT(strlen(CONFIG_BT_OTS_DIR_LIST_OBJ_NAME) <= CONFIG_BT_OTS_OBJ_MAX_NAME_LEN,
+		 "BT_OTS_DIR_LIST_OBJ_NAME shall be less than or equal to %u octets",
+		 CONFIG_BT_OTS_OBJ_MAX_NAME_LEN);
+
+#if defined(CONFIG_BT_OTS_OBJ_NAME_WRITE_SUPPORT)
+	static char dir_list_obj_name[CONFIG_BT_OTS_OBJ_MAX_NAME_LEN + 1];
```

Comment:
This function is not thread-safe now.

Why do we need this if it is just using a copy of CONFIG_BT_OTS_DIR_LIST_OBJ_NAME anyway?

Reply:
In order to be able to modify the name, it needs to be backed by mutable memory but `CONFIG_BT_OTS_DIR_LIST_OBJ_NAME` could be held in nonmutable memory. In the case when `CONFIG_BT_OTS_OBJ_NAME_WRITE_SUPPORT` is enabled the mutable name variable is initialized from `CONFIG_BT_OTS_DIR_LIST_OBJ_NAME` and then can be modified by a client.

Ideally I would have just left this alone but enabling the name write support means that all object names including the built in directory object name is modifiable.

Very good point on the thread safety issue here though. Your comment actually has pointed out that the current implementation would be sharing the directory object's name between different OTS instances which is not ok. I have updated the implementation with one that has an object name per OTS instance if the `CONFIG_BT_OTS_OBJ_NAME_WRITE_SUPPORT` is set.

Reply:
There's nothing in the spec that prohibits the server from reject a write to the name characteristic as far as I know, so checking if the current object is the directory listing object and then rejecting the write should be fine. 

I am not sure I follow the thread safe issue: Isn't there only a single thread (the RX thread) that will call the write callback?

Reply:
Oh, I was under the impression that based on [this comment](https://github.com/zephyrproject-rtos/zephyr/pull/35996#discussion_r648201187) we don't want to reject an object name write. If we can do that, then I will change the implementation to reject writes to the directory object's name.

I think the thread safety issue was only during the initialization when multiple ots instances could be initialized but the would be accessing a single object name instance. If the name is remaining nonmutable, I don't think there is an issue anymore.